osu.Game/Screens/Play/HUD/HealthDisplay.cs

```diff
@@ -29,12 +32,18 @@ protected virtual void Flash(JudgementResult result)
         {
         }
 
-        [BackgroundDependencyLoader]
-        private void load()
+        [BackgroundDependencyLoader(true)]
+        private void load(HUDOverlay hud)
         {
             Current.BindTo(HealthProcessor.Health);
 
             HealthProcessor.NewJudgement += onNewJudgement;
+
+            if (hud != null)
+            {
+                showHealthbar = hud.ShowHealthbar.GetBoundCopy();
+                showHealthbar.BindValueChanged(healthBar => this.FadeTo(healthBar.NewValue ? 1 : 0, HUDOverlay.FADE_DURATION, HUDOverlay.FADE_EASING), true);
```

Comment:
That's a `BindValueChanged(..., true)` in BDL - I'm not sure whether oversight or intentional, though, given this becomes a bit uglier if the callback was to be conditionally invoked in `LoadComplete()`? Doesn't crash on the transforms at least.

Reply:
While it's probably safe as both components are added together, but I'll move it anyways.